src/java.base/share/classes/java/lang/Object.java

```diff
@@ -33,6 +33,25 @@
  * Every class has {@code Object} as a superclass. All objects,
  * including arrays, implement the methods of this class.
  *
+ * @apiNote
+ * <h2><a id=objectOverrides>Guidance for overriding {@code Object}
+ * methods</a></h2>
+ *
+ * {@code java.lang.Object} includes three {@code public} methods
+ * overridable in subclasses, {@link #equals(Object)}, {@link
+ * #hashCode()}, and {@link #toString()}. Each of these methods can be
+ * called by any code that has access to an object of a
+ * class. Especially if an object can be used by code in a separate
+ * maintenance domain, any overrides of these methods should take care
+ * to provide reasonable implementations of these methods. In
+ * particular, the implementations should take care to avoid using
+ * excessive memory, computational time, or any other resources.
```

Comment:
While it's not wrong, advising against "excessive" usage is so vague as not to be actionable. Given some piece of code it's hard to say whether or not it does anything excessive. Consider a List with a million elements; as a practical matter, any one of these methods requires visiting every element. Is that excessive? Why or why not?

Reply:
> While it's not wrong, advising against "excessive" usage is so vague as not to be actionable. Given some piece of code it's hard to say whether or not it does anything excessive. Consider a List with a million elements; as a practical matter, any one of these methods requires visiting every element. Is that excessive? Why or why not?

Hmm. In rare cases, I think providing not-quite-actionable advice is on net helpful.

My basic goal here is to convey "when you're implementing these methods, be considerate to your expected users." If that results in some extra contemplation during development or code review about whether or not the worst-case behavior of hashCode/toString/equals is excessive, I think that would be a reasonable outcome from this change.